app/models/answer.rb

```diff
@@ -0,0 +1,37 @@
+class Answer < ActiveRecord::Base
+  belongs_to :question
+  belongs_to :user
+  has_many :responses, as: :respondable, dependent: :destroy
+  has_many :votes, as: :votable, dependent: :destroy
+
+  def editable_by? user
+    self.user == user
+  end
+
+  def vote_sum
+    Vote.where("votable_type = ? AND votable_id =?", "Answer", self.id).pluck("value").inject(0){|sum,x| sum + x }
+  end
+
+  def already_voted?(user_id)
+    !Vote.where("votable_type = ? AND user_id = ? AND votable_id = ?", "Answer", user_id, self.id).empty?
```

Comment:
`votes.where(user_id: user_id).count > 0` will return a boolean as to whether this user has voted.